File: ContactSurvey/src/main/java/com/verint/xm/contactSurvey/MainActivity.java
package com.verint.xm.contactSurvey;

import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;

import com.verint.xm.sdk.Core;
import com.verint.xm.sdk.SurveyManagement;


public class MainActivity extends AppCompatActivity {
	
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        
        // Do normal UI setup
        setContentView(R.layout.main);

        // Back button
        getSupportActionBar().setDisplayHomeAsUpEnabled(false);
    }

    @Override
    protected void onResume() {
        super.onResume();

        // Launch an invite as a demo if we're eligible
        SurveyManagement.checkIfEligibleForSurvey();
    }

    public void launchInvite(View view) {
        // Increment the significant event count so that we're eligible for an invite
        // based on the criteria in foresee_configuration.json
        SurveyManagement.incrementSignificantEventCountWithKey("instant_invite");

        // Launch an invite as a demo if we're eligible
        SurveyManagement.checkIfEligibleForSurvey();
    }

    public void resetCounters(View view) {
    	// Reset the state of the ForeSee SDK
    	Core.resetState();
    }

    public void setContactDetails(View view) {
        startActivity(new Intent(this, SetContactDetailsActivity.class));
    }
}

File: KotlinSample/src/main/java/com/verint/xm/kotlinsample/MainActivity.kt
package com.verint.xm.kotlinsample

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.View
import com.verint.xm.sdk.Core
import com.verint.xm.sdk.SurveyManagement

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main)
    }


    fun checkEligibility(@Suppress("UNUSED_PARAMETER") view: View) {
        // Increment the significant event count so that we're eligible for an invite
        // based on the criteria in foresee_configuration.json
        SurveyManagement.incrementSignificantEventCountWithKey("instant_invite")

        // Launch an invite as a demo
        SurveyManagement.checkIfEligibleForSurvey()
    }

    fun resetCounters(@Suppress("UNUSED_PARAMETER") view: View) {
        // Reset the state of the ForeSee SDK
        Core.resetState()
    }
}
